src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -36,7 +36,11 @@ internal New3Command(INewCommandInput commandInput, ITemplateEngineHost host, IT
         {
             _telemetryLogger = telemetryLogger;
             _host = new CliTemplateEngineHost(host, commandInput);
-            EnvironmentSettings = new EngineEnvironmentSettings(_host, settingsLocation: hivePath, virtualizeSettings: virtualize);
+            EnvironmentSettings = new EngineEnvironmentSettings(
+                _host,
+                settingsLocation: hivePath,
+                virtualizeSettings: virtualize,
+                environment: new DefaultEnvironment(new Dictionary<string, string>() { { "TEMPLATE_ENGINE_DISABLE_FILEWATCHER", "1" } }));
```

Comment:
let's go for custom `IEnvironment` implementation as discussed.

Reply:
Created new CliEnviroment.